api/utils/keys/yubikey.go

```diff
@@ -227,27 +246,42 @@ func (y *YubiKeyPrivateKey) Public() crypto.PublicKey {
 
 // Sign implements crypto.Signer.
 func (y *YubiKeyPrivateKey) Sign(rand io.Reader, digest []byte, opts crypto.SignerOpts) ([]byte, error) {
-	signCtx, cancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
 	// To prevent concurrent calls to Sign from failing due to PIV only handling a
 	// single connection, use a lock to queue through signature requests one at a time.
 	y.signMux.Lock()
 	defer y.signMux.Unlock()
 
-	yk, err := y.open()
+	signature, err := y.sign(ctx, rand, digest, opts)
+	if err != nil && strings.Contains(err.Error(), pivAuthErrMessage) {
```

Comment:
If the message is only checked here, I'd move the const here. I'd also add a comment that it was acquired from an actual PIV key.